aspnetcore/mvc/extensibility/app-parts.md

```diff
@@ -0,0 +1,124 @@
+---
+title: Application Parts | Microsoft Docs
+author: ardalis
+description: 
+keywords: ASP.NET Core,
+ms.author: riande
+manager: wpickett
+ms.date: 1/4/2017
+ms.topic: article
+ms.assetid: b355a48e-a15c-4d58-b69c-899963613a98
+ms.technology: aspnet
+ms.prod: aspnet-core
+uid: mvc/extensibility/app-parts
+---
+# Application Parts
+
+By [Steve Smith](http://ardalis)
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/mvc/extensibility/app-parts/sample)
+
+An *Application Part* is a resource where MVC features may be discovered, such as an assembly. *Feature providers* work with application parts to populate the features of an ASP.NET Core MVC app. 
```

Comment:
An Application Part is an abstraction over the resources of an application, like types, from where MVC features like controllers, view components or tag helpers are discovered. One example of an application part is an AssemblyPart, which encapsulates an assembly reference and exposes types and compilation references.